var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -180,10 +180,13 @@ def yes_no(varstr):
             options += ['MPICXX=%s' % spec['mpi'].mpicxx]
 
         if '+hypre' in spec:
-            options += [
-                'HYPRE_DIR=%s' % spec['hypre'].prefix,
-                'HYPRE_OPT=-I%s' % spec['hypre'].prefix.include,
-                'HYPRE_LIB=-L%s' % spec['hypre'].prefix.lib + ' -lHYPRE']
+            hypre = spec['hypre']
+            hp = hypre.prefix
+            options += ['HYPRE_DIR=%s' % hp,
+                        'HYPRE_OPT=-I%s' % hp.include,
+                        'HYPRE_LIB=-L%s' % (hp.lib + ' -lHYPRE ' + ' '.join(
```

Comment:
i know it's not directly related, but you might want to clean-up `hypre` and add custom `libs()` there to search for `lHYPRE` library and use it here. Then you can add all three and call for `ld_flags`.

Reply:
I've submitted a separate PR with this behavior in #7234. Once it lands, I can rewrite the link flags using `spec['hypre'].libs.ld_flags`.